src/lib/sitemap/helpers.ts

```diff
@@ -0,0 +1,36 @@
+import { SitemapElement, SitemapFactoryInput, SitemapPriority } from './types'
+
+/**
+ *
+ * @param slug
+ * @returns 1 - 0.7 depending on slug depth in relation to a product's base path
+ *
+ * See XML Tag Definitions: https://www.sitemaps.org/protocol.html
+ */
+export const determinePriority = (slug: string): SitemapPriority => {
+	const slugDepth = slug.split('/').length
+	switch (slugDepth) {
+		case 1:
+			return 1
+		case 2:
+			return 0.9
+		case 3:
+			return 0.8
+		default:
+			return 0.7
+	}
+}
+
+export const makeSitemapField = ({
+	slug,
+	lastmodDate,
+	priority,
+	changefreq,
+}: SitemapFactoryInput): SitemapElement => {
+	return {
+		loc: `${__config.dev_dot.canonical_base_url}${slug}`,
```

Comment:
💡 Not totally necessary for this PR but you could use the URL api to ensure that the pathname is formatted properly. If not, you might want to use a regex to remove leading and trailing slashes or something.

Reply:
Added!